python/lsst/ts/tunablelaser/csc.py

```diff
@@ -0,0 +1,185 @@
+"""Implements CSC classes for the TunableLaser.
+
+"""
+from .component import LaserComponent
+from lsst.ts import salobj
+import asyncio
+import pathlib
+from lsst.ts.idl.enums import TunableLaser
+
+
+class LaserCSC(salobj.ConfigurableCsc):
+    """This is the class that implements the TunableLaser CSC.
+
+    Parameters
+    ----------
+    address : `str`
+        The physical usb port string where the laser is located.
+    frequency : `float`, optional
+        The amount of time that the telemetry stream is published.
+    initial_state : `salobj.State`, optional
+        The initial state that a CSC will start up in. Only useful for unit
+        tests as most CSCs will start in `salobj.State.STANDBY`
+
+    Attributes
+    ----------
+    model : `LaserModel`
+        The model of the laser component which handles the actual hardware.
+
+    """
+    def __init__(self, index,
+                 initial_state=salobj.State.STANDBY, config_dir=None, initial_simulation_mode=0):
+        schema_path = pathlib.Path(__file__).resolve().parents[4].joinpath("schema", "TunableLaser.yaml")
+        super().__init__(name="TunableLaser", index=index, schema_path=schema_path, config_dir=config_dir,
+                         initial_state=initial_state, initial_simulation_mode=initial_simulation_mode)
+        self.model = LaserComponent()
+        self.evt_detailedState.set_put(detailedState=TunableLaser.DetailedState.NONPROPAGATING)
+
+    async def telemetry(self):
+        """Send out the TunableLaser's telemetry.
+
+        """
+        try:
+            while True:
+                if self.summary_state is not salobj.State.FAULT:
+                    try:
+                        self.model.publish()
+                    except TimeoutError:
+                        self.fault(code=TunableLaser.LaserErrorCode.timeout_error)
+                    if (self.model.CPU8000.power_register.register_value == "FAULT"
+                        or self.model.M_CPU800.power_register.register_value == "FAULT"
+                        or self.model.M_CPU800.power_register_2.register_value == "FAULT") \
+                       and self.summary_state is not salobj.State.FAULT:
+                        self.fault(code=TunableLaser.LaserErrorCode.hw_cpu_error,
+                                   report=(f"Code:{self.CPU8000.fault_register.fault}"
+                                           f" Code:{self.M_CPU800.fault_register.fault}"
+                                           f" Code:{self.M_CPU800.fault_register_2.fault}"))
```

Comment:
I suggest you return after calling self.fault(...) (always a good policy) and get rid of the `if` on the next line.